### PR TITLE
DRILL-3610: Functional tests for TIMESTAMPADD and TIMESTAMPDIFF

### PR DESCRIPTION
**Pull request by abhidotravi**
- Basic functional tests for `TIMESTAMPADD` and `TIMESTAMPDIFF` functions.
- Tests cover different `SQL_TSI_` intervals.

### CODE REVIEW

File: framework/resources/Functional/timestamp_functions/timestamp_functions.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      "schema": "dfs.drilltestdirtpch01parquet",
+      "output-format": "tsv",
+      "expected-file": ".*.e_tsv",
+      "verification-type": [
+        "in-memory"
+      ]
+    }
+  ],
+  "datasources": [
+  ]
```
Comment (Agirish) [inline framework/resources/Functional/timestamp_functions/timestamp_functions.json]: Hey, I understand you are reusing an existing dataset (TPC-H 0.01). But we recommend each test def file to be self sufficient and independent. So can you look up one of the other test suites which uses this dataset / workspace and provide that in the datasources section?

Comment (aravi5) [reply framework/resources/Functional/timestamp_functions/timestamp_functions.json]: I actually referred to `Functional/filter/selectivity/plan` and `Functional/filter/selectivity/negative/plan` to build the definition file and they didn't seem to have anything mentioned for `datasources`.

Comment (aravi5) [reply framework/resources/Functional/timestamp_functions/timestamp_functions.json]: Done.

Comment (Agirish) [reply framework/resources/Functional/timestamp_functions/timestamp_functions.json]: Yeah, I think we might have missed catching it, when adding that suite. But in general, we would want to be able to run any test suite independently. Thanks for fixing this.

**Pull request merged**